src/Orchard.Cms.Web/Modules/Orchard.Diagnostics/Startup.cs

```diff
@@ -21,6 +25,21 @@ public override void Configure(IApplicationBuilder app, IRouteBuilder routes, IS
             // c.f. https://docs.asp.net/en/latest/fundamentals/error-handling.html#id3
             app.UseStatusCodePagesWithReExecute("/Error", "?status={0}");
 
+            app.Use((context, next) =>
+            {
+                string contentType;
+                if (_contentTypeProvider.TryGetContentType(context.Request.Path, out contentType))
+                {
+                    var statusCodePagesFeature = context.Features.Get<IStatusCodePagesFeature>();
+                    if (statusCodePagesFeature != null)
+                    {
+                        statusCodePagesFeature.Enabled = false;
+                    }
+                }
+
+                return next();
```

Comment:
You could put the code after the `next()` call, and then check for an error status code before even checking the content type.